### ISSUE TITLE
Add an ability to use environment variables for login/password

### ISSUE THREAD

**Issue by alivespirit** [ashwanthkumar/gocd-slack-build-notifier#46]
Hi,
We need to have an ability to use environment variables for the login/password in config file - something like "login = ${GO_USERNAME}". We need this because we want to grant users access to config file under source control - so they could add their own pipelines notifications for their own channels. I've tried any possible way I could for using environment variables in config, but no luck - plugin always fails to start with the messages like:

```
FelixGoPluginOSGiFramework:124 - Failed to load plugin: plugins_work/gocd-slack-notifier-1.2.2.jar. Plugin is invalid. Reasons [Class [GoNotificationPlugin] is annotated with @Extension bu
t cannot be constructed. Reason: com.typesafe.config.ConfigException$NotResolved: need to Config#resolve(), see the API docs for Config#resolve(); substitution not resolved: ConfigReference(${GO_USERNAME})., No extensions found in
 this plugin.Please check for @Extension annotations]
```

Can you please check if there are any possible options for us?

**Comment by ashwanthkumar** [comment]
@alivespirit Typesafe config does support resolving environmental variables. I'll probably send a patch sometime tonight. This should be easy to add.

**Comment by ashwanthkumar** [comment]
@alivespirit From the [documentation](https://github.com/typesafehub/config/tree/v1.2.1#optional-system-or-env-variable-overrides) you can see that the environment variables are already supported out of the box[1]. Can you please check that when starting the go server you've the `GO_USERNAME` environment variable is present? Probably you can cross verify it on `cat /proc/<pid of go-server.jar>/environ`. 

[1] - Actual test case - https://github.com/typesafehub/config/blob/v1.2.1/config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala#L730-L762

**Comment by alivespirit** [comment]
@ashwanthkumar Sorry for the late reply.
Those variables were added to `/etc/go/go-server.config`, which is exported by go-server init.d script. I have checked and `GO_USERNAME` and `GO_PASSWORD` variables are present in environ file. 
Also, relevant config section looks like this:

```
  login = ${?GO_USERNAME}
  password = ${?GO_PASSWORD}
```

When using quotes, no errors are mentioned in log, but on any pipelines actions there are errors 401 unauthorized in slack notifier logs, so it is not substituting vars with the quotes around.
